app/components/Shared/CustomSlider.vue

```diff
@@ -0,0 +1,53 @@
+<script setup lang="ts">
+import { Carousel, Navigation, Slide ,Pagination} from 'vue3-carousel'
+import 'vue3-carousel/dist/carousel.css'
+import type { Props } from '~/types/slider'
+
+const props = withDefaults(defineProps<Props>(), {
+  breakpoints: () => ({
+    700: {
+      itemsToShow: 3.5,
+      snapAlign: 'center'
+    },
+    1024: {
+      itemsToShow: 5,
+      snapAlign: 'start'
+    }
+  }),
+  products: () => [],
+  showPagination: true,
+  showNavigation: true,
+  wrapAround: false,
+  autoplay: 1
+})
+// Define the settings and breakpoints using `ref` since they are reactive
+const settings = ref({
+  itemsToShow: 1,
+  snapAlign: 'center'
+})
+
+
+</script>
+<template>
+  <Carousel :autoplay="props.autoplay" v-bind="settings" :breakpoints="props.breakpoints" :wrap-around="props.wrapAround">
+    <Slide v-for="(slide ,index) in products" :key="slide.id||index">
+      <div class="carousel__item">
+        <slot :slide="slide" ></slot>
+
+      </div>
+    </Slide>
+   
+    <template #addons>
+      <Pagination />
+
+    <navigation v-if="props.showNavigation">
```

Comment:
(In general)
`props.` prefix is not needed in the template.

```suggestion
    <navigation v-if="showNavigation">
```